### DISCUSSION TITLE
Unable to run pipeline-runner image under docker on amd64 system

### DISCUSSION THREAD

**Discussion by bpow** [broadinstitute/seqr#3069] [Q&A] [answered]
Not sure that this issue should be filed here or with the [seqr-loading-pipelines](https://github.com/broadinstitute/seqr-loading-pipelines) project, but when working through the [LOCAL_INSTALL.md](https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md) document I run into issues with docker-compose trying to start the `pipeline-runner` container.

When running `docker-compose up -- pipeline-runner`, I get:

```
elasticsearch is up-to-date
Starting seqr_pipeline-runner_1 ... done
Attaching to seqr_pipeline-runner_1
pipeline-runner_1  | exec /entrypoint.sh: exec format error
seqr_pipeline-runner_1 exited with code 1
```

Running `docker run  -it gcr.io/seqr-project/pipeline-runner:gcloud-prod` gives a little more info:

```
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
exec /entrypoint.sh: exec format error
```

Running `docker run --platform linux/amd64 -it gcr.io/seqr-project/pipeline-runner:gcloud-prod` doesn't help, the error then includes the following:

```
docker: Error response from daemon: image with reference gcr.io/seqr-project/pipeline-runner:gcloud-prod was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64.
```

*However*, the `:latest` tag, on the other hand (instead of `:gcloud-prod`), does have the linux/amd64 platform.

I don't know if this difference is intended (e.g., is it intended to run only on arm64 on gcloud?), or accidental (did a developer build on an M1 mac and push/tag). Either way, it would be nice for the LOCAL_INSTALL.md file to refer to an available docker image:tag which can be run in the same platform as the other images.

**Comment by hanars** [comment]
As discussed [here](https://github.com/broadinstitute/seqr/discussions/3031#discussioncomment-4110355), amd64 is not a supported platform. I'm not sure why the latest tag works for that, but I would not rely on that continuing to work. In the linked discussion are instructions for build your own image for the platform

**Comment by bpow** [comment]
OK, so I'm not the only one to have run into this :).

I'm fine with building my own image, but I think it would be helpful, if the `arm64` architecture is the only one supported for this step, that this be mentioned in the `LOCAL_INSTALL.md` document under the requirements. It may be that this is not the case when using "Option #1" from that document, but following through "Option #2" in that document, there is nothing that indicates that we need to use a different platform for the pipeline-runner container than for the seqr instance.

Also, the platform of the default pipeline-runner instance looks to have changed just in August of this year-- prior to that, the images being made had amd64 arch:

```bash
for i in 20221019_171650 20221007_143623 20220829_165356 20220526_130649 20220526_125536 20220428_114713 20220413_114830 20220318_162331 20220216_152748 20220126_144423 20220126_144230 20220125_165333
do
echo -n "$i "
docker inspect gcr.io/seqr-project/pipeline-runner:$i | grep Arch
done
```

Gives:
```
20221019_171650         "Architecture": "arm64",
20221007_143623         "Architecture": "arm64",
20220829_165356         "Architecture": "arm64",
20220526_130649         "Architecture": "amd64",
20220526_125536         "Architecture": "amd64",
20220428_114713         "Architecture": "amd64",
20220413_114830         "Architecture": "amd64",
20220318_162331         "Architecture": "amd64",
20220216_152748         "Architecture": "amd64",
20220126_144423         "Architecture": "amd64",
20220126_144230         "Architecture": "amd64",
20220125_165333         "Architecture": "amd64",
```

So, I think it would still be helpful at the least to address this more directly in the `LOCAL_INSTALL.md` document, or even better to ask the pipeline-runner team to produce a separately-tagged image for the same arch that you use for `gcr.io/seqr-project/seqr:gcloud-prod` so that it is possible to run the `docker-compose.yml` from a single machine without use of a vm for the separate arch.

**Comment by hanars** [comment, marked as answer]
You are correct, I did not actually realize that we changed platform support a couple months ago. That was actually a bug, and I have now released an updated image build for amd64. Sorry about that, and thanks for your helpful insight!
